azkaban-common/src/main/java/azkaban/executor/container/watch/AzPodStatusMetadata.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2021 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.executor.container.watch;
+
+import static azkaban.executor.container.KubernetesContainerizedImpl.CLUSTER_LABEL_NAME;
+import static azkaban.executor.container.KubernetesContainerizedImpl.EXECUTION_ID_LABEL_NAME;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.util.Watch;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is intended for maintaining any relevant data from a Pod Watch event that was derived
+ * during the inference of {@link AzPodStatus}. The data here can be consumed by any downstream
+ * callbacks.
+ */
+public class AzPodStatusMetadata {
```

Comment:
Shall we rename this class to AzPodEventMetadata?

Reply:
The name `AzPodStatusMetadata` was intended to convey the fact that this contains 'more' data associated with the simple enums defined in `AzPodStatus`. 
Let me know if you have a strong reason for the rename (I do like the sound of of `AzPodEventMetadata` as well).

Reply:
For future extensibility, I feel event metadata will be little generic. No hard core reason though.

Reply:
Retaining the current name in that case, will revisit in a subsequent PR if this needs to be changed. Thanks!